nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -143,18 +155,35 @@ public Batch getBatch(String batchId) {
 
         // create the batch
         Batch batch = new Batch(batchId, parameters, getName(), getTransientStore());
-        AssumeRoleRequest assumeRoleRequest = new AssumeRoleRequest().withRoleSessionName(batch.getKey())
-                                                                     .withRoleArn(roleArn);
-        Credentials credentials = stsClient.assumeRole(assumeRoleRequest).getCredentials();
+
+        String bucketPath = concatPath(bucketPrefix, batchId) + "/";
```

Comment:
`concatPath` remove the last `/` and you add it back here, why ?

Reply:
Also `concatPath` is way overkill as it's only called with two arguments. Why not just do

    String bucketPath = bucketPrefix + batchId + '/';

There's no reason to second-guess the slashes in the bucket prefix. It's a prefix added to all names, that's it.